### PR TITLE
Selenium

### PR DESCRIPTION
**Pull request by madhura0404**

### CODE REVIEW

File: selenium/6.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+Selenium::WebDriver::Chrome::Service.driver_path = 'C:\Users\MADHURA\Desktop\Git training\selenium\drivers\chromedriver.exe'
+driver= Selenium::WebDriver.for :chrome
+
+driver.get "https://letcode.in/forms"
+
+dropdown = driver.find_element(:xpath, "/html/body/app-root/app-forms/section[1]/div/div/div[1]/div/div/form/div[5]/div[2]/div/div/div/select")
+
+puts dropdown.text
+
+dropdown.click
```
Comment (joyrel-vaz) [inline selenium/6.rb:20]: This step is not required

File: selenium/6.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require 'selenium-webdriver'
+
+Selenium::WebDriver::Chrome::Service.driver_path = 'C:\Users\MADHURA\Desktop\Git training\selenium\drivers\chromedriver.exe'
+driver= Selenium::WebDriver.for :chrome
+
+driver.get "https://letcode.in/forms"
+
+dropdown = driver.find_element(:xpath, "/html/body/app-root/app-forms/section[1]/div/div/div[1]/div/div/form/div[5]/div[2]/div/div/div/select")
+
+puts dropdown.text
```
Comment (joyrel-vaz) [inline selenium/6.rb]: A better approach is to use

`dropdown = Selenium::WebDriver::Support::Select.new(dropdown_xpath_element)`

You should make use of the functions mentioned here: https://www.selenium.dev/selenium/docs/api/rb/Selenium/WebDriver/Support/Select.html

File: selenium/9.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+require 'selenium-webdriver'
+
+Selenium::WebDriver::Chrome::Service.driver_path = 'C:\Users\USER\Desktop\Desktop\Git training\selenium\drivers\chromedriver.exe'
+driver= Selenium::WebDriver.for :chrome
+
+
+driver.get "https://cosmocode.io/automation-practice-webtable/"
+
+table_header= driver.find_elements(:tag_name, 'h3')
```
Comment (joyrel-vaz) [inline selenium/9.rb]: `h3` is not a good locator to get table header. Use `tr` instead.

File: selenium/10.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+column_index = 3 
+   
+rows = table.find_elements(:tag_name, 'tr')
+
+rows.each do |row|
+
+    columns = row.find_elements(:tag_name, 'td')
+    
+    if columns.size > column_index
```
Comment (joyrel-vaz) [inline selenium/10.rb:18]: Why is this check done?

File: selenium/11.rb [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+row_index = 2    
+column_index = 3 
+
+rows = table.find_elements(:tag_name, 'tr')
+if row_index < rows.size
+    row = rows[row_index]
+    
+    cells = row.find_elements(:tag_name, 'td')
+    
+    if column_index < cells.size
```
Comment (joyrel-vaz) [inline selenium/11.rb]: Why is this check done?

File: selenium/11.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+driver= Selenium::WebDriver.for :chrome
+
+driver.get 'https://the-internet.herokuapp.com/tables'  
+  
+table = driver.find_element(:id, 'table1')  
+row_index = 2    
+column_index = 3 
+
+rows = table.find_elements(:tag_name, 'tr')
+if row_index < rows.size
```
Comment (joyrel-vaz) [inline selenium/11.rb:13]: Why is this check done?

File: selenium/12.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+select.select_by(:value, "za")
+
+add_to_cart = driver.find_element(:xpath, '//*[@id="inventory_container"]/div/div[4]/div[3]/button')
+add_to_cart.click
+
+cart = driver.find_element(:id,'shopping_cart_container')
+cart.click
+
+checkout= driver.find_element(:xpath,'//*[@id="cart_contents_container"]/div/div[2]/a[2]')
+checkout.click
```
Comment (joyrel-vaz) [inline selenium/12.rb]: Code can be modularised into methods performing different test flows

**Review by joyrel-vaz — CHANGES_REQUESTED** [review]

File: selenium/11.rb (hunk 5 above)
Comment (madhura0404) [reply selenium/11.rb]: it is done to check if the row_index and column_index is not greater than the no. of rows / columns in the table.

File: selenium/11.rb (hunk 6 above)
Comment (madhura0404) [reply selenium/11.rb:13]: it is done to check if the row_index and column_index is not greater than the no. of rows / columns in the table.

File: selenium/11.rb (hunk 5 above)
Comment (madhura0404) [reply selenium/11.rb]: But i think I should have used 'else' to print error

File: selenium/11.rb (hunk 6 above)
Comment (madhura0404) [reply selenium/11.rb:13]: But i think I should have used 'else' to print error

File: selenium/10.rb (hunk 4 above)
Comment (madhura0404) [reply selenium/10.rb:18]: it is done to check if the no. of columns in the table is greater than the column_index .

**Pull request closed without merging**